### PR TITLE
intel_adsp/ace: power: No pending transaction before power gate

### PR DESCRIPTION
**Pull request by ceolin**
Issue an upstream read transaction through uncached memory to flush out all pending transactions before power down the host domain.

Note: It also fixes a problem in the host power domain when using only one core.

### CODE REVIEW

File: soc/intel/intel_adsp/ace/power.c [hunk 1]
```diff
@@ -25,6 +25,14 @@
 #define SRAM_ALIAS_BASE         0xA0000000
 #define SRAM_ALIAS_MASK         0xF0000000
 
+#if CONFIG_SOC_INTEL_ACE15_MTPM
+/* Used to force any pending transaction by HW issuing an upstream read before
+ * power down host domain.
+ */
+uint32_t g_key_read_holder __aligned(CONFIG_DCACHE_LINE_SIZE);
+uint8_t g_key_read_holder_bottom_alignment[CONFIG_DCACHE_LINE_SIZE - sizeof(uint32_t)];
```
Comment (andyross) [inline soc/intel/intel_adsp/ace/power.c:33]: Looks like this _bottom_alignment array is unused?   What's it for?   If the goal is to force padding for g_key_read_holder, that doesn't look like it's going to work.  We build with -fdata-sections and the linker will throw it out, leaving something arbitrary in that cache line.

Really just make the whole thing an aligned byte array of DCACHE_LINE_SIZE, you're going to cast it to a word pointer below anyway.

**Review by andyross — COMMENTED** [review]
Oh, and while you're refactoring this code, it would be pleasing to fix the unfortunate hungarian notation and give the variable a Zephyrish scoped name like "intel_adsp_ace_power_cookie" or whatever.

File: soc/intel/intel_adsp/ace/power.c (hunk 1 above)
Comment (ceolin) [reply soc/intel/intel_adsp/ace/power.c:33]: indeed, thanks for pointing it out.

**Review by andyross — APPROVED** [review]

**Review by andyross — APPROVED** [review]

**Review by tmleman — APPROVED** [review]

**Review by kv2019i — APPROVED** [review]
Took me a while to understand the commit as this is modifying existing code to do the same. So commit explain what is done, but the actual code changes only update the implementation. If you otherwise update the PR, maybe reword the commit msg as well.

**Review by nashif — APPROVED** [review]
